### PR TITLE
Add extra unit tests

### PR DESCRIPTION
**Pull request by bancer**

### CODE REVIEW

File: lib/Cake/Test/Case/Controller/ApplicationControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ * @return CakeResponse|null
+ */
+	public function next() {
+		$sessionName = session_name();
+		$sessionId = $this->Session->id();
+		return $this->redirect(array(
+			'controller' => 'trans_session_id',
+			'action' => 'next_step',
+			'?' => array(
+				htmlspecialchars($sessionName) => htmlspecialchars($sessionId),
```
Comment (markstory) [inline lib/Cake/Test/Case/Controller/ApplicationControllerTest.php]: Why are you HTML escaping URL parameters? The Location header isn't vulnerable to XSS. CakePHP will handle urlencoding this data.

File: lib/Cake/Test/Case/Controller/ApplicationControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+				'session.use_cookies' => 0,
+				'session.use_only_cookies' => 0,
+			),
+			'defaults' => 'php',
+			'cookieTimeout' => 240,
+			'cacheLimiter' => 'must-revalidate',
+		);
+		$actualConfig = Configure::read('Session');
+		$this->assertEquals($expectedConfig, $actualConfig);
+		$this->assertTrue(false, __METHOD__ . ' test has been run');
```
Comment (markstory) [inline lib/Cake/Test/Case/Controller/ApplicationControllerTest.php]: We can't merge this while it is failing.

Comment (bancer) [reply lib/Cake/Test/Case/Controller/ApplicationControllerTest.php]: Right, I have not finished yet.

File: lib/Cake/Test/Case/Controller/ApplicationControllerTest.php (hunk 1 above)
Comment (bancer) [reply lib/Cake/Test/Case/Controller/ApplicationControllerTest.php]: Right, that's not important in this case.

**Comment by bancer** [comment]
@markstory This test is risky when run with PHP unit 5.7.19. Somehow that is related to  `$ini['session.use_trans_sid'] = 1;`.

**Comment by bancer** [comment]
```
1) ApplicationControllerTest::testRedirect
Test code or tested code did not (only) close its own output buffers
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestCase.php:84
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestRunner.php:75
/home/travis/build/cakephp/cakephp/lib/Cake/TestSuite/CakeTestSuiteCommand.php:98
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:283
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Command/TestShell.php:268
/home/travis/build/cakephp/cakephp/lib/Cake/Console/Shell.php:462
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:222
/home/travis/build/cakephp/cakephp/lib/Cake/Console/ShellDispatcher.php:66
```

@markstory It seems there is a bug in `CakeTestCase::run` method. Perhaps something related to `ob_start()`. I have difficulties in figuring out what is wrong there :(

**Comment by bancer** [comment]
@chinpei215 maybe you could take a look at this?

**Comment by bancer** [comment]
@markstory I think it was not ready for merging. There is still a problem that is not fixed - `Test code or tested code did not (only) close its own output buffers`.

**Comment by markstory** [comment]
@bancer Ok. I can revert the merge.

**Pull request merged**